app/services/support_interface/external_report_candidates_export.rb

```diff
@@ -0,0 +1,268 @@
+            raw_data.status,
+            COUNT(*)
+        FROM
+            raw_data
+        GROUP BY
+            raw_data.sex,
+            raw_data.area,
+            raw_data.age_group,
+            raw_data.status
+      SQL
+    end
+
+    def deferred_offers_counts
+      @deferred_offers_counts ||= ActiveRecord::Base
+        .connection
+        .execute(deferred_offers_query)
+        .to_a
+    end
+
+    def deferred_offers_query
+      <<-SQL
+        WITH raw_data AS (
+            SELECT
+                c.id,
+                f.id,
+                CASE
+                  WHEN ac.status_before_deferral = 'recruited' THEN 'Recruited'
+                  WHEN ac.status_before_deferral = 'pending_conditions' THEN 'Conditions pending'
+                END status_before_deferral,
+                CASE
+                  #{equality_and_diversity_sql}
+                END sex,
+                CASE
+                  #{area_sql}
+                END area,
+                CASE
+                  #{age_group_sql}
+                END age_group
+            FROM
+                application_forms f
+            LEFT JOIN
+                candidates c ON f.candidate_id = c.id
+            LEFT JOIN
+                application_choices ac ON ac.application_form_id = f.id
+            WHERE
+                NOT c.hide_in_reporting
+                AND f.recruitment_cycle_year = #{RecruitmentCycle.previous_year}
+                AND f.date_of_birth IS NOT NULL
+                AND ac.status_before_deferral IS NOT NULL
+                AND f.region_code IN (
+                  'eastern',
+                  'east_midlands',
+                  'london',
+                  'north_east',
+                  'north_west',
+                  'northern_ireland',
+                  'scotland',
+                  'south_east',
+                  'south_west',
+                  'wales',
+                  'west_midlands',
+                  'yorkshire_and_the_humber',
+                  'european_economic_area',
+                  'rest_of_the_world'
+                )
+            GROUP BY
+                c.id, f.id, ac.status_before_deferral
+        )
+        SELECT
+            raw_data.status_before_deferral,
+            raw_data.sex,
+            raw_data.area,
+            raw_data.age_group,
+            COUNT(*)
+        FROM
+            raw_data
+        GROUP BY
+            raw_data.status_before_deferral, raw_data.sex, raw_data.area, raw_data.age_group
+      SQL
+    end
+
+    def equality_and_diversity_sql
+      "WHEN f.equality_and_diversity is NULL THEN 'Not provided'
+      WHEN f.equality_and_diversity->> 'sex' = 'intersex' then 'Intersex'
+      WHEN f.equality_and_diversity->> 'sex' = 'male' then 'Male'
+      WHEN f.equality_and_diversity->> 'sex' = 'female' then 'Female'
+      WHEN f.equality_and_diversity->> 'sex' = 'Prefer not to say' then 'Prefer not to say'"
+    end
+
+    def area_sql
```

Comment:
So the main reason i did the mappings this way is because i assumed (maybe incorrectly) that it would be quite a bit more performant. Do you think it won't matter too much?